builds/git_version.mk

```diff
@@ -0,0 +1,10 @@
+GIT_ROOT = $(REPO_ROOT)/.git
+GIT_HEAD = $(GIT_ROOT)/HEAD
+GIT_REV_FILE = $(shell $(REPO_ROOT)/builds/git_ref.sh $(GIT_HEAD))
+
```

Comment:
Can you explain why all of this magic is needed? [`git rev-parse`](https://git-scm.com/docs/git-rev-parse) will do exactly what is needed here.



fine tuned: `git rev-parse --verify -q --short=10 HEAD^{commit}`

Reply:
We can and probably should use git to generate the string. However we need to figure out which file contains the hash and add it as a dependency of the rule. This ensures that `git_version.cpp` is regenerated whenever `HEAD` changes.

Reply:
`git rev-parse --absolute-git-dir`
`git symbolic-ref -q HEAD`
?